src/test/test-process-util.c

```diff
@@ -379,7 +379,7 @@ static void test_rename_process_now(const char *p, int ret) {
 
         assert_se(get_process_cmdline(0, 0, false, &cmdline) >= 0);
         /* we cannot expect cmdline to be renamed properly without privileges */
-        if (geteuid() == 0) {
+        if (geteuid() == 0 && !running_in_userns()) {
```

Comment:
`test-process-util` fails to rename its command line because of [the lack the `CAP_SYS_RESOURCE` capability in the initial namespace](https://github.com/torvalds/linux/blob/ead751507de86d90fa250431e9990a8b881f713c/kernel/sys.c#L2046), so `running_in_userns` seems to be too broad and I don't think the patch should be merged as is.

Probably `rename_process` should be changed so as to signal whether it has been successful or not so that `test_rename_process` could decide what to do.

Reply:
@evverx I think we do need a generic check function for "if have this capability in the initial namespace" because we are continuously hitting bugs w.r.t. so and so namespace being available in the current namespace, but not the initial, and kernel checking for the initial namespace instead of current.

Reply:
@xnox such a check would be very handy indeed, but I'm not sure it's possible. Do you know how to do that?